### DISCUSSION TITLE
Field for values in a nested dictionary

### DISCUSSION THREAD

**Discussion by JustBennnn** [pydantic/pydantic#3246] [Question] [answered]
Is there a way I can create a field for values inside of a nested dictionary?

```python
dictionary = {"a": 5, "b": {"c": 6}}
class Test(BaseModel):
    a: int
    c: ?
```

How do I define `c` so I can access it with `Test(**dictionary).c`?

**Comment by modaye** [comment, marked as answer]
from pydantic import *


class Test(BaseModel):
    a: int
    c: int

    class Config:
        extra = Extra.allow

    @root_validator(pre=True)
    def parse_variables(cls, values):
        for k in list(values['b']):
            values[k] = values['b'][k]
        del values['b']
        return values


data = {"a": 5, "b": {"c": 6}}
assert Test(**data).dict() == {'a': 5, 'c': 6}
assert Test(**data).c == 6

**Reply by JustBennnn** [reply]
Thanks for the reply!
In the end, I did find another way to achieve this:
```python
class Test(BaseModel):
    a: int
    b_object: Dict[str, int] = Field(alias="b)

    class Extra(BaseModel):
        c: int
    
    @property 
    def c(self):
        return self.Extra(**self.b_object)

data = {"a": 5, "b": {"c": 6}}
c = Test(**data).c
```
